src/handlers/commands.js

```diff
@@ -46,36 +46,101 @@ module.exports = async (message, gdb, db, countingChannel, prefix) => {
   if (message.channel.id == countingChannel) setTimeout(() => message.channel.id == gdb.get().channel ? deleteMessages([ message, response ]) : null, 5000);
 };
 
-module.exports.setupSlashCommands = async client => {
+module.exports.setupSlashCommands = async (client, db) => {
   client.ws.on("INTERACTION_CREATE", async interaction => {
     const commandFile = commands.get(interaciton.data.name);
     if (!commandFile) return respondWithError(interaction, `❌ The command \`${interaction.data.name}\` does not exist.`)
 
     const guild = client.guilds.cache.get(interaction_guild_id);
     if (!guild) return respondWithError(interaction, `❌ This guild is currently unavailable.`)
     const channel = guild.channels.cache.get(interaction.channel_id);
-    if (!channel) return respondWithError(interaction, `❌ I don't have access to view this channel.`)
+    if (!channel || !(commandFile.needAccessToChannel && !channel.viewable)) return respondWithError(interaction, `❌ I don't have access to view this channel.`)
```

Comment:
Ln 57 remove negation of `(commandFile.needAccessToChannel && !channel.viewable)`